typing/typecore.ml

```diff
@@ -1569,37 +1747,35 @@ and type_pat_aux
       | Some r, (_ :: _ as exs)  ->
           let exs = List.map (Ctype.existential_name constr) exs in
           let name = constr.cstr_name in
-          raise (Error (loc, !env, Unexpected_existential (r,name, exs)))
+          raise (Error (loc, !env, Unexpected_existential (r, name, exs)))
       end;
-      (* if constructor is gadt, we must verify that the expected type has the
-         correct head *)
-      if constr.cstr_generalized then
-        unify_head_only ~refine loc env (instance expected_ty) constr;
-      let sarg' =
+      let sarg', svto =
```

Comment:
out of curiosity, what does `vto` mean?

Reply:
(variable * type) option
Could be more explicit.

Reply:
Renamed to `existential_{styp,ctyp}`.